github/github.go

```diff
@@ -336,7 +336,11 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 		return nil, err
 	}
 
-	defer resp.Body.Close()
+	defer func() {
+		// Drain and close the body to let the Transport reuse the connection
+		io.Copy(ioutil.Discard, resp.Body)
```

Comment:
interestingly, I just tried this, and it's apparently discard 0 bytes, yet I'm still seeing the same speed improvements @FiloSottile is reporting.  Maybe it's just the io.EOF that is left to be read which is preventing the transport from being reused?